### PR TITLE
rbd: add disk usage tool

### PR DESCRIPTION
**Pull request by dillaman**
The new disk usage tool uses the new fast diff object map feature
(when enabled) to quickly calculate the provisioned vs actual usage
of images and associated snapshots within a pool.

Fixes: #7746
Signed-off-by: Jason Dillaman <dillaman@redhat.com>

### CODE REVIEW

File: src/rbd.cc [hunk 1]
```diff
@@ -2550,6 +2553,198 @@ static int parse_map_options(char *options)
...
+                                    TextTable& tbl, Formatter *f,
+                                    uint64_t *used_size) {
+  const char* from = NULL;
+  if (!from_snap_name.empty()) {
+    from = from_snap_name.c_str();
+  }
+
+  uint64_t flags;
+  int r = image.get_flags(&flags);
+  if (r < 0) {
```
Comment (jdurgin) [inline src/rbd.cc]: should probably accept the method not existing on osds here (or change librbd's get_flags() http://tracker.ceph.com/issues/11553)

Comment (dillaman) [inline src/rbd.cc]: @jdurgin librbd's get_flags() just pulls the current state of the flags from ImageCtx.  The only time the ```get_flags``` OSD method is invoked is during an image refresh and that is protected w/ EOPNOTSUPP/EIO checks.

**Comment by jdurgin** [comment]
Looks good. Could you add a test to make sure we don't change this reporting accidentally? Maybe one in test/cli-integration

**Comment by dillaman** [comment]
@jdurgin CLI formatting test added

**Pull request merged**